libsel4/tools/rustgenerator.py

```diff
@@ -0,0 +1,615 @@
+#
+# SPDX-License-Identifier: BSD-2-Clause
+#
+
+from generator import (
+    MAX_MESSAGE_LENGTH,
+    Generator,
+    is_result_struct_required,
+    struct_members,
+    get_parameter_positions
+)
+
+
+class RustGenerator(Generator):
+    @staticmethod
+    def translate_type(name):
+        type_trans = {
+            "int": "isize",
+            "seL4_Uint8": "u8",
+            "seL4_Uint16": "u16",
+            "seL4_Uint32": "u32",
+            "seL4_Uint64": "u64",
+            "seL4_Bool": "u8",
+            "seL4_CapData_t": "seL4_CapData",
+            "seL4_PrioProps_t": "seL4_PrioProps",
+            "seL4_CapRights_t": "seL4_CapRights",
+            "seL4_RISCV_Page_GetAddress_t": "seL4_RISCV_Page_GetAddress",
+        }
+        if name in type_trans:
+            return type_trans[name]
+        else:
+            return name
+
+    @staticmethod
+    def translate_expr(name):
+        # TODO there are cases which implement this in-line, refactor to use this function instead
+        #      or get rid of this
+        if name == "type":
+            return "type_"
+        return name
+
+    TYPES = {
```

Comment:
I wonder if instead of using the plain numbers, having anther indirection here might one day pay off. It might be abstraction overkill also...